email_alert_service/models/email_alert.rb

```diff
@@ -43,4 +44,24 @@ def email_api_client
   def strip_empty_arrays(tag_hash)
     tag_hash.reject { |_, tags| tags.empty? }
   end
+
+  def document_links
+    strip_empty_arrays(
+      document.fetch("links", {}).merge("taxon_tree" => taxon_tree)
+    )
+  end
+
+  def taxon_tree
+    return [] unless document.dig("links", "taxons")
+
+    # TODO: update the public API of the taxonomy helpers gem to also accept a
+    # single fully expanded document so that we don't have to pass in the
+    # document twice.
+    linked_content_item = GovukTaxonomyHelpers::LinkedContentItem.from_publishing_api(
+      content_item: document,
+      expanded_links: document,
```

Comment:
@tijmenb I replied to your comment on https://github.com/alphagov/rummager/pull/762#discussion_r106369670, I think it's a similar problem.

I'd like to think about this more, but I probably won't be able to look at it until next week.

IMO there are two main use cases at the moment

- Given a content item, tell me about it's taxons and the taxons above them
- Given a taxon, access the whole taxonomy

The second case is still quite fiddly because get_expanded_links doesn't tell you about the taxon you're on, but I think we simplify the first one.